pilot/pkg/serviceregistry/kube/controller/endpoints.go

```diff
@@ -121,11 +120,12 @@ func (e *endpointsController) InstancesByPort(c *Controller, svc *model.Service,
 		return nil
 	}
 	ep := item.(*v1.Endpoints)
+	hostname := kube.ServiceHostname(ep.Name, ep.Namespace, e.c.domainSuffix)
 	var out []*model.ServiceInstance
 	for _, ss := range ep.Subsets {
 		for _, ea := range ss.Addresses {
 			var podLabels labels.Instance
-			pod := c.pods.getPodByIP(ea.IP)
+			pod, _ := getPod(c, ea.IP, &ep.ObjectMeta, ea.TargetRef, hostname)
```

Comment:
Calling this function may have side effects of queueEndpointEventOnPodArrival which is unexpected here

Reply:
Yeah, i realized that, will look into if there is another way.

Reply:
After a close look and analysis, there is no harm for queueEndpointEventOnPodArrival twice. Actually, it can be called many times before. But i can create another function to get ride of `queueEndpointEventOnPodArrival ` though.
With regard to the issue: pod comes later than endpoint, i do not change anything. It seems not correct if we can not get the pod here. We need pod labels/serviceacccount/workloadname. If these are missing, the intermediate xds configuration maybe not right. I will file an issue here.

Reply:
@howardjohn take a look at the legacy bug https://github.com/istio/istio/issues/29405